src/system_data.jl

```diff
@@ -1142,6 +1142,47 @@ function get_available_associated_components(
     return components
 end
 
+"""
```

Comment:
@jd-lara @GabrielKS I now consider the interface finalized. I decided against adding a filter function. We could add one with the signature `function x(::InfrastructureSystemsComponent, ::SupplementalAttribute)::Bool`. Seems like overkill to me, but I'm willing to add it if you think people will want it.